tests/pages/newsletter_embed.py

```diff
@@ -0,0 +1,107 @@
+        # Ideally here we would focus on the email field, but Selenium has issues
+        # dealing with focus events when the window is not active.
+        # See bug https://bugzilla.mozilla.org/show_bug.cgi?id=704583
+        self.root.find_element(*self._submit_button_locator).click()
+        Wait(self.selenium, self.timeout).until(
+            expected.visibility_of_element_located(self._privacy_policy_checkbox_locator))
+
+    @property
+    def is_form_detail_displayed(self):
+        return self.root.find_element(*self._form_details_locator).is_displayed()
+
+    @property
+    def email(self):
+        el = self.root.find_element(*self._email_locator)
+        return el.get_attribute('value')
+
+    def type_email(self, value):
+        self.root.find_element(*self._email_locator).send_keys(value)
+
+    @property
+    def country(self):
+        el = self.root.find_element(*self._country_locator)
+        return el.find_element(By.CSS_SELECTOR, 'option[selected]').text
+
+    def select_country(self, value):
+        el = self.root.find_element(*self._country_locator)
+        Select(el).select_by_visible_text(value)
+
+    @property
+    def html_format_selected(self):
+        el = self.root.find_element(*self._html_format_locator)
+        return el.is_selected()
+
+    def select_html_format(self):
+        self.root.find_element(*self._html_format_locator).click()
+
+    @property
+    def text_format_selected(self):
+        el = self.root.find_element(*self._text_format_locator)
+        return el.is_selected()
+
+    def select_text_format(self):
+        self.root.find_element(*self._text_format_locator).click()
+
+    @property
+    def privacy_policy_accepted(self):
+        el = self.root.find_element(*self._privacy_policy_checkbox_locator)
+        return el.is_selected()
+
+    def accept_privacy_policy(self):
+        el = self.root.find_element(*self._privacy_policy_checkbox_locator)
+        assert not el.is_selected(), 'Privacy policy has already been accepted'
+        el.click()
+        assert el.is_selected(), 'Privacy policy has not been accepted'
+
+    def click_sign_me_up(self):
+        self.root.find_element(*self._submit_button_locator).click()
+        Wait(self.selenium, self.timeout).until(
+            expected.visibility_of_element_located(self._thank_you_locator))
+
+    @property
+    def sign_up_successful(self):
+        el = self.selenium.find_element(*self._thank_you_locator)
+        return el.is_displayed()
+
+
+class MozillaNewsletterEmbedForm(NewsletterEmbedForm):
+
+    _root_locator = (By.ID, 'mozorg-newsletter-form')
+    _country_locator = (By.ID, 'country')
+    _success_url_slug = 'sign-up-for-mozilla'
+
+    def click_sign_me_up(self):
+        self.root.find_element(*self._submit_button_locator).click()
+        Wait(self.selenium, self.timeout).until(
+            lambda s: self._success_url_slug in s.current_url)
```

Comment:
We could now change this to: `Wait(self.selenium, self.timeout).until(lambda s: self.sign_up_successful)`